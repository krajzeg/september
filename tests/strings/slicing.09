string := "012345"

print("This slice should print '12':", string[1..2])
print("This should be the same:", string[1...3])
print("Out of bounds indices should be OK:", string[4..10])
print("If both indices are out of bounds, the result should be empty:", string[20..24])
print("If from is greater than to, the slice should be empty:", string[7..5])
print("The string[1..end] syntax should work, so this is 12345:", string[1..end])
print("Negative indices within slices should work, so this is 234:", string[(-4)..(-2)])

