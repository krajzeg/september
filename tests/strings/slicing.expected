This slice should print '12': 12
This should be the same: 12
Out of bounds indices should be OK: 45
If both indices are out of bounds, the result should be empty: 
If from is greater than to, the slice should be empty: 
The string[1..end] syntax should work, so this is 12345: 12345
Negative indices within slices should work, so this is 234: 234
